File: apps/kira-studio/internal/codeparse/testdata/extract/sample.java
interface Greeter {
    String greet(String name);
}

class Person implements Greeter {
    public String greet(String name) {
        return helper(name);
    }

    private String helper(String name) {
        return "hi " + name;
    }
}

class Robot extends Person {
    void build() {
        Person p = new Person();
    }
}

File: apps/kira-studio/internal/codeparse/testdata/extract/sample.rs
trait Greeter {
    fn greet(&self, name: &str) -> String;
}

struct Person;

impl Greeter for Person {
    fn greet(&self, name: &str) -> String {
        helper(name)
    }
}

fn helper(name: &str) -> String {
    format!("hi {}", name)
}
